### PR TITLE
docs: refactor the directory, add typedoc and hosted on ghpages

### PR DESCRIPTION
**Pull request by yorkie**
> Note: this is an early stage, and don't **merge** this.

I refactored our documentation site in the followings:

- Using [docsify](https://docsify.js.org/) to refactor the whole documents which supports markdowns, better multi-languages and search.
- Using typedoc to generate the API even though the current edition is hard to read, because of our structure and comments.
- Deployment to GitHub Pages which combine the above parts.

An online preview is at: https://yorkie.github.io/pipcook/#/

@utkobe @wordcount Please take a review on:

- The new TOC.
- The new way to write documents and language versions, it means we'll move to GitHub to update documentations.
- The new website UX.
- The new workflow.

### CODE REVIEW

**Comment by yorkie** [comment]
I have not updated the Chinese version completely for now, please ignore them if something went wrong there.

**Comment by utkobe** [comment]
@yorkie There are some links directing to the wrong place. (e.g. links in [Plugin Specification](https://yorkie.github.io/pipcook/#/tutorials/developer-guide?id=plugin-specifications) leads to 404. I guess docsify generates wrong url at this time?

File: docs/tutorials/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+We have defined a set of interfaces for each plugin. Each type of plugin must be implemented strictly according to the interfaces. The detailed information is as follows:
+
+- [Data Collect](./plugin/0-data-collect.md)
+- [Data Access](./plugin/1-data-access.md)
+- [Data Process](./plugin/2-data-process.md)
+- [Model Load](./plugin/3-model-load.md)
+- [Model Train](./plugin/4-model-train.md)
+- [Model Evaluate](./plugin/5-model-evaluate.md)
+- [Model Deploy](./plugin/6-model-deploy.md)
```
Comment (yorkie) [inline docs/tutorials/developer-guide.md]: > @yorkie There are some links directing to the wrong place. (e.g. links in Plugin Specification leads to 404. I guess docsify generates wrong url at this time?

Good catch, it's not a bug of docsify, the URIs here are wrong by me, should be "../spec/plugin/....", I will fix them now.

Comment (yorkie) [reply docs/tutorials/developer-guide.md]: Fixed at baa7870.

**Comment by utkobe** [comment]
Cool, now the English version seems ok to me. Just wait for Chinese one to be completed. @yorkie

**Comment by yorkie** [comment]
> Cool, now the English version seems ok to me. Just wait for Chinese one to be completed. @yorkie

I will update tonight, thank you :)

**Comment by yorkie** [comment]
Updated done, @utkobe @wordcount Please have a complete review, and it seems to be merge-able :)

**Comment by utkobe** [comment]
> Updated done, @utkobe @wordcount Please have a complete review, and it seems to be merge-able :)

Hyperlinks in this [page](https://yorkie.github.io/pipcook/#/zh-cn/devel/developer-guide) seems not correct, still pointing to old github pages site. Could you have a check please

**Comment by yorkie** [comment]
> > Updated done, @utkobe @wordcount Please have a complete review, and it seems to be merge-able :)
> 
> Hyperlinks in this [page](https://yorkie.github.io/pipcook/#/zh-cn/devel/developer-guide) seems not correct, still pointing to old github pages site. Could you have a check please

Thanks for your eyes, fixed at my latest commit :)

**Comment by utkobe** [comment]
> > > Updated done, @utkobe @wordcount Please have a complete review, and it seems to be merge-able :)
> > 
> > 
> > Hyperlinks in this [page](https://yorkie.github.io/pipcook/#/zh-cn/devel/developer-guide) seems not correct, still pointing to old github pages site. Could you have a check please
> 
> Thanks for your eyes, fixed at my latest commit :)

Sorry, just found also this [page](https://yorkie.github.io/pipcook/#/zh-cn/spec/plugin). And after fixing this, everything is good for me. @wordcount can have a double check please.

**Comment by yorkie** [comment]
@utkobe Fixed.

**Pull request merged**